core/trino-main/src/test/java/io/trino/sql/planner/sanity/TestValidateScaledWritersRightnessUsage.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.sanity;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.trino.connector.CatalogName;
+import io.trino.connector.MockConnectorFactory;
+import io.trino.execution.warnings.WarningCollector;
+import io.trino.metadata.TableHandle;
+import io.trino.plugin.tpch.TpchColumnHandle;
+import io.trino.plugin.tpch.TpchTableHandle;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.sql.PlannerContext;
+import io.trino.sql.planner.Partitioning;
+import io.trino.sql.planner.PartitioningScheme;
+import io.trino.sql.planner.PlanNodeIdAllocator;
+import io.trino.sql.planner.Symbol;
+import io.trino.sql.planner.TypeProvider;
+import io.trino.sql.planner.assertions.BasePlanTest;
+import io.trino.sql.planner.iterative.rule.test.PlanBuilder;
+import io.trino.sql.planner.plan.PlanNode;
+import io.trino.sql.planner.plan.TableScanNode;
+import io.trino.testing.LocalQueryRunner;
+import io.trino.testing.TestingTransactionHandle;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.trino.SessionTestUtils.TEST_SESSION;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.sql.planner.SystemPartitioningHandle.SCALED_WRITER_DISTRIBUTION;
+import static io.trino.sql.planner.SystemPartitioningHandle.SINGLE_DISTRIBUTION;
+import static io.trino.sql.planner.TypeAnalyzer.createTestingTypeAnalyzer;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestValidateScaledWritersRightnessUsage
```

Comment:
Does this test have to be single threaded?

Reply:
is `planBuilder` immutable or threadsafe?

Reply:
It is not, there is some shared state. And, yes, it means that test should be SingleThreaded (same for all tests using `PlanBuilder` (if they do not introduce synchronization)